### ISSUE TITLE
RFC make `persistStore` return `persistor` in a promise

### ISSUE THREAD

**Issue by mattkrick** [rt2zz/redux-persist#124]
First, love the repo, it's magic!

I've got 2 sources of data: 1 comes down from the server, and another is stored locally. I get both together & then start the app (in this case, call my router).

Putting all my router logic in the callback function works, but I'd love it if the `persistor` was returned in a promise, after all, it's not like we could really use it before the data is rehydrated anyways, right?

That way, we could get rid of the `callback` and just write something like:
`const persistor = await persistStore()`

if folks don't use the `persistor`, then there's no breaking change.

**Comment by rt2zz** [comment]
The challenge is, sometimes it is nice to get the persistor synchronously, e.g. if you want to call `.purge()`

I would want to think hard about any other possible use cases for having the persistor immediately before making this change. Open to more thoughts on the matter..

**Comment by mattkrick** [comment]
I completely agree `purge` is a great method, my only thought is that you'd never call `purge` before `rehydrate` completed, right? Otherwise it's possible that you continue to rehydrate after a purge (assuming the rehydration is done in multiple async chunks).

If it's a promise, it guarantees there's no race condition between `rehydrate` and `purge`.

**Comment by darsain** [comment]
👍 the current API is awkward to use and makes the code ugly. I see no reason why I'd want to purge the store while it's loading. I would assume that would break something.

What actually might be useful is a way to only purge with no loading (i.e. purge previously saved state). That is not supported right now as far as I'm aware. The API would look something like this:

```js
const persistor = persistStore(store, config); // drop the callback, and don't auto-load

// explicit load
persistor.load(callback); // calls the callback
persistor.load();         // returns promise

// fully controlled purge, can be called before or after loading
persistor.purge();

**Comment by rt2zz** [comment]
you can purge the persistor at any time with either `purgeStoredState(config)` or `persistor.purge()`.

Not starting the persistor by default is an interesting idea.
